apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -36,6 +36,44 @@ namespace Google.Cloud.Spanner.V1
     /// </summary>
     public sealed partial class SessionPool
     {
+        /// <summary>
+        /// This struct contains information required to identify the database sessions.
+        /// It will be used as key in <see cref="_targetedPools"/>.
+        /// </summary>
+        internal readonly struct SessionPoolInfo
+        {
+            /// <summary>
+            /// Database name of the associated <see cref="TargetedSessionPool"/>.
+            /// </summary>
+            internal DatabaseName DatabaseName { get; }
+
+            /// <summary>
+            /// Database role of the associated <see cref="TargetedSessionPool"/>.
+            /// </summary>
+            internal string DatabaseRole { get; }
+
+            internal SessionPoolInfo(DatabaseName databaseName, string dataabaseRole)
+            {
+                DatabaseName = databaseName;
+                DatabaseRole = dataabaseRole;
+            }
+
+            /// <inheritdoc/>
+            public override string ToString() => $"Database Name = {DatabaseName};Database role = {DatabaseRole}";
```

Comment:
Nit: I'd add a space after the semi-colon in the string, and probably change the "key" parts to the property names, so `$"DatabaseName = {DatabaseName}; DatabaseRole = {DatabaseRole}"`